smqtk_detection/impls/detect_image_objects/centernet.py

```diff
@@ -129,155 +133,220 @@ def __init__(
         self.input_h = 960
         self.input_w = 1280
 
+        self.out_h = self.input_h // 2
+        self.out_w = self.input_w // 2
+
         self.num_classes = 10
 
         # visdrone2019 mean and standard deviation, used for normalization
         self.mean = np.asarray([[[0.372949, 0.37837514, 0.36463863]]], dtype=np.float32)
         self.std = np.asarray([[[0.19171683, 0.18299586, 0.19437608]]], dtype=np.float32)
 
-        if use_cuda and torch.cuda.is_available():
-            self.device = torch.device('cuda')
+        if use_cuda:
+            if torch.cuda.is_available():
+                self.device = torch.device('cuda')
+            else:
+                logger.info("CUDA device not available, using CPU.")
+                self.device = torch.device('cpu')
         else:
             self.device = torch.device('cpu')
 
         self.model = _CenterNet(resnet_spec[self.arch], heads)
         self.model.init_weights(model_urls[self.arch], pretrained=True)
         self.model = self.model.to(self.device)
         self.model = _load_model(self.model, model_file)
+        self.model = self.model.eval()
 
     def detect_objects(
         self,
         img_iter: Iterable[np.ndarray]
     ) -> Iterable[Iterable[Tuple[AxisAlignedBoundingBox, Dict[Hashable, float]]]]:
 
-        proc_imgs = []
-        centers = []
-        trans_scales = []
-        out_height = self.input_h // 2
-        out_width = self.input_w // 2
+        img_set = _ImageDataset(img_iter, self.flip, self.scales, self._preprocess_img)
+        batch_loader = DataLoader(img_set, batch_size=self.batch_size)
```

Comment:
It is probably the case that we want to pass something to DataLoader's `num_workers` parameter to make use of the parallelization of data loading. The value to pass should be parameterized in the constructor.

Reply:
Of course, I say that, but I just saw this try to pull over 100GB of RAM so I'm trying to understand that.